src/lib/utils/configureInitialState.js

```diff
@@ -0,0 +1,98 @@
+import moment from 'moment';
+import { DATE_TIME_FORMAT } from '../constants/index';
+
+// Let's define config structure
+const exampleConfig = {
+  repeat: [/* 'Yearly', */'Monthly', 'Weekly', 'Daily', 'Hourly'],
+  yearly: 'on',
+  monthly: 'on the',
+  end: ['On date'],
+};
+
+const configureState = (config = {}) => {
+  const configureFrequency = () => {
+    if (config.repeat) {
+      return config.repeat[0];
+    }
+    return 'Yearly';
+  };
```

Comment:
Consider writing less code i.e. `const configureFrequency = () => config.repeat ? config.repeat[0] : 'Yearly'`

Reply:
It was initially in this form as I didn't know if these functions would extend.
Zipped